### ISSUE TITLE
[esphome] Dallas, rotary_encoder, ht711 and dht not working

### ISSUE THREAD

**Issue by dbuezas** [kuba2k2/libretuya#27]
## Compilation error:
```
undefined reference to `esphome::InterruptLock::InterruptLock()'
```

## Due to it not being defined for LibreTuya
<img width="538" alt="image" src="https://user-images.githubusercontent.com/777196/193578899-8be94d89-bcc1-422a-b040-4a5ed06ddf10.png">

## Solution
`#elif defined(USE_ESP32) || defined(USE_LIBRETUYA)`

**Comment by dbuezas** [comment]
Dallas (and others probably too) use micros() inside interruptLock blocks, which is also broken.

**Comment by kuba2k2** [comment]
Fixed in #28.
